exonum/src/runtime/mod.rs

```diff
@@ -12,109 +12,110 @@
+//! only exception is the genesis block. The object hashes of the built-in services are included
+//! into the resulting state hash of the genesis block. This behavior difference will be unified in
+//! the following release.
 //!
-//! [`Dispatcher`] is responsible for persisting artifacts and services across node restarts.
+//! The [`Dispatcher`] is responsible for persisting artifacts and services across node restarts.
 //!
 //! # Transaction Lifecycle
 //!
-//! 1. An Exonum client creates a transaction message which includes [`CallInfo`] information
-//!   about the corresponding method to call and serialized method parameters as a payload.
+//! 1. An Exonum client creates a transaction message which includes two parts. The first part is
+//!   the [`CallInfo`] information about the corresponding method to call. The second part is the
+//!   serialized method parameters as a payload.
 //!   The client then signs the message using the Ed25519 signature system.
 //!
 //! 2. The client transmits the message to one of the Exonum nodes in the network.
 //!
 //! 3. The node verifies correctness of the transaction signature and retransmits it to
 //!   the other network nodes if it is correct.
 //!
-//! 4. When the consensus algorithm run by Exonum nodes finds a feasible candidate for the next block
-//!   of transactions, transactions in the block are passed to the [`Dispatcher`] for execution.
+//! 4. When the consensus algorithm finds a feasible candidate for the next block
+//!   of transactions, transactions in this block are passed to the [`Dispatcher`] for execution.
 //!
-//! 5. The dispatcher uses a lookup table to find the corresponding [`Runtime`] for the transaction
-//!   by the [`instance_id`] recorded in the message. If the corresponding runtime is
-//!   found, the dispatcher passes the transaction into this runtime for
-//!   immediate [execution].
+//! 5. The dispatcher uses a lookup table to find the corresponding [`Runtime`] for each transaction
+//!   by the [`instance_id`] recorded in the transaction message. If the corresponding runtime exists,
+//!   the dispatcher passes the transaction into this runtime for immediate [execution].
 //!
 //! 6. After execution the transaction [execution status] is written into the blockchain.
 //!
 //! # Supervisor Service
 //!
 //! A supervisor service is a service that has additional privileges. This service
-//! allows to deploy artifacts and instantiate new services after the blockchain is launched
+//! allows deploying artifacts and instantiating new services after the blockchain is launched
 //! and running. Other than that, it looks like an ordinary service.
-//! The supervisor should be present during the blockchain start, otherwise no new artifacts / services
-//! could ever be added to the blockchain.
+//! To enable adding new artifacts / services to the blockchain after its start, the supervisor
+//! must be present in the system binary file during the blockchain start.
```

Comment:
I suppose this is technically correct with the current supervisor implementation, but AFAIK there are no obstacles preventing a non-Rust supervisor implementation. Thus, I'd suggest to reform to

> [...] the supervisor must be one of the builtin service instances.

Possibly, it's worth it specifying above what a builtin service instance is.